src/components/ApplyView.tsx

```diff
@@ -0,0 +1,297 @@
+          <Button variant="secondary" size="sm" onClick={handleReject}>
+            <XIcon className="mr-1 h-3 w-3" />
+            Reject All
+          </Button>
+          <Button size="sm" onClick={handleAccept}>
+            <Check className="mr-1 h-3 w-3" />
+            Apply Changes
+          </Button>
+        </div>
+      </div>
+
+      <div className="flex-1 overflow-auto p-2">
+        {changeBlocks.map((block, blockIndex) => {
+          // Check if this block contains any changes (added or removed)
+          const hasChanges = block.some((change) => change.added || change.removed);
+
+          // Check if all changes in this block are accepted or rejected
+          const isAccepted = hasChanges && block.every((change) => change.accepted);
+          const isRejected = hasChanges && block.every((change) => change.rejected);
+
+          return (
+            <div
+              key={blockIndex}
+              className={cn(
+                "mb-4 border rounded-md overflow-hidden transition-colors duration-200",
+                {
+                  "border-green shadow-[0_0_5px_rgba(var(--color-green-rgb),0.2)]": isAccepted,
+                  "border-red shadow-[0_0_5px_rgba(var(--color-red-rgb),0.2)]": isRejected,
+                }
+              )}
+            >
+              {block.map((change, changeIndex) => (
+                <div
+                  key={`${blockIndex}-${changeIndex}`}
+                  className={cn("flex relative border-l-3", {
+                    "bg-modifier-success border-l-green": change.added,
+                    "bg-modifier-error border-l-red": change.removed,
+                    "opacity-50":
+                      (change.added && change.rejected) || (change.removed && !change.accepted),
+                  })}
+                >
+                  <div className="w-6 flex-shrink-0 flex items-center justify-center text-sm font-bold bg-background-secondary-alt">
+                    {change.added && "+"}
+                    {change.removed && "-"}
+                  </div>
+                  <div className="flex-1 font-mono text-sm whitespace-pre-wrap py-1 px-2 text-text-normal">
+                    {change.value}
+                  </div>
+                </div>
+              ))}
+
+              {/* Only show accept/reject buttons for blocks with changes */}
+              {hasChanges && (
+                <div className="flex justify-end p-2 bg-background-secondary-alt border-t border-border gap-2">
+                  <Button variant="secondary" size="sm" onClick={() => rejectBlock(blockIndex)}>
+                    <XIcon className="h-4 w-4" />
+                    Reject
+                  </Button>
+                  <Button size="sm" onClick={() => acceptBlock(blockIndex)}>
+                    <Check className="h-4 w-4" />
+                    Accept
+                  </Button>
+                </div>
+              )}
+            </div>
```

Comment:
can we apply the change directly in this view when a user accept or reject a block? It's not very clear after a user accepts and rejects the change and what's going to happen